app/main.py

```
@@ -1,14 +1,61 @@
-class Validator:
-    pass
+from abc import ABC, abstractmethod
+from typing import Any
 
 
-class Number:
-    pass
+class Validator(ABC):
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = f"_{name}"
 
+    @abstractmethod
+    def validate(self, value: Any) -> None:
+        pass
 
-class OneOf:
-    pass
+    def __get__(self, instance: Any, owner: type) -> Any:
+        return getattr(instance, self.protected_name, None)
```

Validator.__get__ must return the descriptor object when accessed on the class (instance is None). The current implementation uses `getattr(instance, self.protected_name, None)` and will return `None` for class access (because `instance` is None) and also silently return `None` if the protected attribute is not present on the instance. Change the method to explicitly handle class access and to return the stored attribute for instances, e.g.:```python
def __get__(self, instance, owner):
    if instance is None:
        return self
    return getattr(instance, self.protected_name)
```
This matches the descriptor protocol required by the task and ensures `BurgerRecipe.buns` returns the descriptor object. (See checklist items about __get__ behavior.)